pkg/modules/services/implservices/module.go

```diff
@@ -0,0 +1,527 @@
+	if err != nil {
+		return nil, err
+	}
+
+	// Prepare phase
+	queryRangeReq, startMs, endMs, err := m.buildQueryRangeRequest(req)
+	if err != nil {
+		return nil, err
+	}
+
+	// Fetch phase
+	resp, err := m.executeQuery(ctx, orgUUID, queryRangeReq)
+	if err != nil {
+		return nil, err
+	}
+
+	// Process phase
+	items, serviceNames := m.mapQueryRangeRespToServices(resp, startMs, endMs)
+	if len(items) == 0 {
+		return nil, nil
+	}
+
+	// attach top level ops to service items
+	if len(serviceNames) > 0 {
+		if err := m.attachTopLevelOps(ctx, serviceNames, startMs, items); err != nil {
+			return nil, err
+		}
+	}
+
+	return items, nil
+}
+
+// GetTopOperations implements services.Module for QBV5 based top ops
+func (m *module) GetTopOperations(ctx context.Context, orgID string, req *servicetypesv1.TopOperationsRequest) ([]servicetypesv1.TopOperationItem, error) {
+	if req == nil {
+		return nil, errors.NewInvalidInputf(errors.CodeInvalidInput, "request is nil")
+	}
+
+	orgUUID, err := valuer.NewUUID(orgID)
+	if err != nil {
+		return nil, err
+	}
+
+	qr, err := m.buildTopOpsQueryRangeRequest(req)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := m.executeQuery(ctx, orgUUID, qr)
+	if err != nil {
+		return nil, err
+	}
+
+	items := m.mapTopOpsQueryRangeResp(resp)
+	return items, nil
+}
+
+// GetEntryPointOperations implements services.Module for QBV5 based entry point ops
+func (m *module) GetEntryPointOperations(ctx context.Context, orgID string, req *servicetypesv1.EntryPointOperationsRequest) ([]servicetypesv1.EntryPointOperationItem, error) {
+	if req == nil {
+		return nil, errors.NewInvalidInputf(errors.CodeInvalidInput, "request is nil")
+	}
+
+	orgUUID, err := valuer.NewUUID(orgID)
+	if err != nil {
+		return nil, err
+	}
+
+	qr, err := m.buildEntryPointOpsQueryRangeRequest(req)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := m.executeQuery(ctx, orgUUID, qr)
+	if err != nil {
+		return nil, err
+	}
+
+	items := m.mapEntryPointOpsQueryRangeResp(resp)
+	return items, nil
+}
+
+// buildQueryRangeRequest constructs the QBv5 QueryRangeRequest and computes the time window.
+func (m *module) buildQueryRangeRequest(req *servicetypesv1.Request) (*qbtypes.QueryRangeRequest, uint64, uint64, error) {
+	// Parse start/end (nanoseconds) from strings and convert to milliseconds for QBv5
+	startNs, err := strconv.ParseUint(req.Start, 10, 64)
+	if err != nil {
+		return nil, 0, 0, errors.NewInvalidInputf(errors.CodeInvalidInput, "invalid start time: %v", err)
+	}
+	endNs, err := strconv.ParseUint(req.End, 10, 64)
+	if err != nil {
+		return nil, 0, 0, errors.NewInvalidInputf(errors.CodeInvalidInput, "invalid end time: %v", err)
+	}
+	if startNs >= endNs {
+		return nil, 0, 0, errors.NewInvalidInputf(errors.CodeInvalidInput, "start must be before end")
+	}
+	if err := validateTagFilterItems(req.Tags); err != nil {
+		return nil, 0, 0, errors.NewInvalidInputf(errors.CodeInvalidInput, "invalid tags: %v", err)
+	}
```

Comment:
Ideally we would want to keep this validation in json unmarshal but in this case the v1.Request is not directly used so this is fine.